net_utils.js

```diff
@@ -421,3 +428,32 @@ exports.get_ipany_re = function (prefix, suffix, modifier) {
         modifier
     );
 };
+
+exports.ipv6_reverse = function(ipv6){
+    ipv6 = new IPv6.Address(ipv6);
+    return ipv6.reverseForm().replace('.ip6.arpa.', '');
+};
+
+exports.ipv6_bogus = function(ipv6){
+    var ipCheck = new IPv6.Address(ipv6);
+    if(ipCheck.isLinkLocal()) { return true; }
+    if(ipCheck.isMulticast()) { return true; }
+    if(ipCheck.isLoopback()) { return true; }
+    if(ipCheck.isInSubnet(new IPv6.Address('fc00::/7'))) { return true; }
+    if(ipCheck.isInSubnet(new IPv6.Address('ff00::d/8'))) { return true; }
+    var isIPv6Bogus = function(ipCheck){
+        var done = false;
+        var data;
+        var reserveHost = ipCheck.reverseForm().replace('.ip6.arpa.', '')+'.v6.fullbogons.cymru.com.';
+        dns.resolve4(reserveHost, function (err, ip){
+            if(!err) {
+                data = ip[0];
+            }
+            done = true;
+        });
+        deasync.loopWhile( function () { return !done; } );
```

Comment:
Yuck.   Don't like this at all.   Another module dependency and an async function should return a callback IMO.